Formula/luarocks.rb

```diff
@@ -0,0 +1,57 @@
+class Luarocks < Formula
+  desc "Package manager for the Lua programming language"
+  homepage "https://luarocks.org/"
+  url "https://luarocks.org/releases/luarocks-3.0.1.tar.gz"
+  sha256 "b989c4b60d6c9edcd65169e5e42fcffbd39cdbebe6b138fa5aea45102f8d9ec0"
+
+  depends_on "lua"
+  depends_on "lua@5.1" => :recommended
+
+  def install
+    system "./configure", "--prefix=#{prefix}",
+                          "--sysconfdir=#{etc}",
+                          "--rocks-tree=#{HOMEBREW_PREFIX}"
+    system "make", "install"
+  end
+
+  def caveats; <<~EOS
+    LuaRocks supports multiple versions of Lua. By default it is configured
+    to use Lua5.3, but you can require it to use another version at runtime
```

Comment:
By default it's currently configured to require both versions? That seems a bit weird.

Reply:
Should/could one of them be build-time only?

Reply:
I forgot `:test` dependencies are a thing now; I should be able to get away with making `lua@5.1` that.

Reply:
It's not actually configured to use `lua@5.1` during build _at all_; it ignores the fact that it exists in favour of the `lua` formula. I only placed it on the deps list because I wanted to be _really_ thorough about the test here, so I needed access to `lua@5.1`, but forgot there's a less clunky way of achieving that these days.

Reply:
Cool.